tests/test_catalog.py

```diff
@@ -1,30 +1,60 @@
 #!/usr/bin/env python
 
-""" Tests to load the catalog and extend the test capabilities """
-
-__author__ = "A.Radhakrishnan"
-__maintainer__ = "GFDL MSD workflow team"
-
-import intake
-import intake_esm
-import pandas as pd 
-import os
-import pathlib
-
-def load_cat(catspec=None):
-  """Checks if the json and associated csv can be opened by intake_esm""" 
-  cat = None
-  try:
-    cat = intake.open_esm_datastore(catspec)
-  except BaseException as e:
-    print("Can't load this catalog",str(e))
-  return cat
-def test_loadcat():
-  #TODO generate csv on the fly, check if its readable etc
-  catspec = pathlib.Path(os.path.dirname(__file__)).parent / 'cats/gfdl_test1.json'
-  #TODO generate test catalog on the fly, push spec to the test directory
-  cat = load_cat((str(catspec)))
-  try:
-    assert isinstance(cat.df, pd.DataFrame),"test failed"
-  except BaseException as e:
-     assert cat!=None,"opening of esm datastore failed"+str(e)
+import click
+import json
+from jsondiff import diff
+import pandas as pd
+import sys
+
+@click.command()
+@click.argument('json_path', nargs = 1 , required = True)
+@click.argument('json_template_path', nargs = 1 , required = False)
+@click.option('-tf', '--test-failure', is_flag=True, default = False, help="Errors are only printed. Program will not exit.")
```

Comment:
document the purpose of this flag so user can choose to use or ignore.
Sounds like sys.exit is escaped when it comes to the CI usage. 
How does a successful test show up? 

Also suggest renaming tests/config.yaml to a self-described name.  
Wonder why member_id is null, we changed the code to say "na". Was this a made-up test-case?

Reply:
Where do you think it should be documented?

Reply:
The test failure is a temporary solution until the test-case is fixed. Can't remember where it came from.. I copied the existing test config into the config yaml.

Reply:
> Where do you think it should be documented?

Nevermind.. I'll include it in the docstrings

Reply:
Also, a successful test shouldn't return anything.